forms-flow-api/src/formsflow_api/services/application.py

```diff
@@ -506,3 +507,21 @@ def get_application_count(auth, token, **kwargs):
             )
         assert application_count is not None
         return application_count
+    
+    @staticmethod
+    def get_submission_for_application(application_list):
+    # fetch submission data for each application, https://github.com/bcgov/digital-journeys/issues/604
+        applications = []
+        try:
+            formio_service = FormioService()
+            form_io_token = formio_service.get_formio_access_token()
+            for a in application_list:
+                submission = formio_service.get_submission( {"form_id": a["formId"], "sub_id": a["submissionId"]}, form_io_token)
+                if submission:
+                    a["submission"] = submission
+                    applications.append(a)
+        except BusinessException as err:
```

Comment:
I would `raise` the exception in the service and handle it in the application resource.

Reply:
@iman-jamali-fw a thought behind not raising an exception here is. It is s sub-information that we are fetching. The issue with the connection of formio service will break the loading application page.
If we do not raise an exception it will at least load all the applications and the user will not impact here.

I will love to raise an exception if that's the case that seems reasonable. or If you find any point over my thought process please feel free to let me know here,

Reply:
OK, that makes sense.